### ISSUE TITLE
no VCF header and syntax error near unexpected token `if'

### ISSUE THREAD

**Issue by maingoclanmaingoclan** [tseemann/snippy#308]
**> Hello Torsten,
> I have run snippy and then I got errors. I checked the snps.log and the file shows below:**

echo snippy 4.4.0
### bwa index reference/ref.fa

[bwa_index] Pack FASTA... 0.05 sec
[bwa_index] Construct BWT for the packed sequence...
[bwa_index] 3.01 seconds elapse.
[bwa_index] Update BWT... 0.05 sec
[bwa_index] Pack forward-only FASTA... 0.05 sec
[bwa_index] Construct SA from BWT and Occ... 0.66 sec
[main] Version: 0.7.17-r1188
[main] CMD: bwa index reference/ref.fa
[main] Real time: 4.053 sec; CPU: 3.822 sec

### bwa index reference/ref.fa

[bwa_index] Pack FASTA... 0.07 sec
[bwa_index] Construct BWT for the packed sequence...
[bwa_index] 2.91 seconds elapse.
[bwa_index] Update BWT... 0.04 sec
[bwa_index] Pack forward-only FASTA... 0.03 sec
[bwa_index] Construct SA from BWT and Occ... 0.72 sec
[main] Version: 0.7.17-r1188
[main] CMD: bwa index reference/ref.fa
[main] Real time: 5.341 sec; CPU: 3.766 sec

### bwa mem  -Y -M -R '@RG\tID:SNP10\tSM:SNP10' -t 8 reference/ref.fa /home/ubuntu/Anh/Sequences/WTCHG_689876_70045076_1.fastq.gz /home/ubuntu/Anh/Sequences/WTCHG_689876_70045076_2.fastq.gz | samclip --max 10 --ref reference/ref.fa.fai | samtools sort -n -l 0 -T /tmp --threads 8 -m 500M | samtools fixmate -m - - | samtools sort -l 0 -T /tmp --threads 8 -m 500M | samtools markdup -T /tmp -r -s - - > snps.bam

READ 1531521 WRITTEN 1470487
EXCLUDED 6621 EXAMINED 1524900
PAIRED 1491758 SINGLE 33142
DULPICATE PAIR 53488 DUPLICATE SINGLE 7546
DUPLICATE TOTAL 61034

### freebayes-parallel reference/ref.txt 8 -p 2 -P 0 -C 10 --min-repeat-entropy 1.5 --strict-vcf -q 13 -m 60 --min-coverage 10 -F 0.05  -f reference/ref.fa snps.bam > snps.raw.vcf

import: delegate library support not built-in `' (X11) @ error/import.c/ImportImageCommand/1297.
error: no VCF header
/home/ubuntu/miniconda3/bin/vcffirstheader: line 7: syntax error near unexpected token `if'
/home/ubuntu/miniconda3/bin/vcffirstheader: line 7: `    if line.startswith('##'):'
.

> Then I check the vcffirstheader file:

        #!/usr/bin/env python

import sys

header=True
for line in sys.stdin:
    if line.startswith('##'):
        if header:
           print line.strip()
        continue
    elif line.startswith('#'):
        if header:
           print line.strip()
           header=False
        continue
    print line.strip()

> `Could you help me, pl? What should I do now?`

**Comment by tseemann** [comment]
That script is using python 2.x syntax and is expecting the `python` binary to be 2.x, but miniconda has provided `python` executable to be a python 3.x binary.  This needs to be fixed in `bioconda`. I will try and get it resolved ASAP.

Until then, you can edit that file and chance `#!/usr/bin/env python` to `#!/usr/bin/env python2` and see if that works?

**Comment by maingoclanmaingoclan** [comment]
Hello Torsten,
Thanks for your quick response.
I have changed the script but it still got the same error. Is there any other option? Otherwise I will wait until you can fix this.

**Comment by maingoclanmaingoclan** [comment]
Hello Torsten,
I have found out that when I change the script, I did not delete one space, thus its size is 1byte bigger.
So today I  return to the script, I did change again with deletion of one space, and I run again snippy and it works well. But, at the end, there is a message "Wishing you a life free of homopolymer errors." What does this mean? Do I need to worry about it?
I have checked the snps.log file, no error.
Do I need to check anything else to know that the analysis is correct?
Look forward to hearing from you.
Thank you.
